[nnc] Updated indices during broadcast to use int64_t

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64719
* #64716
* __->__ #64627

Differential Revision: [D30801686](https://our.internmc.facebook.com/intern/diff/D30801686)

navahgar committed:
[nnc] Added a call to simplifier before inlining and updated indices during broadcast to use int64_t

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30801686).

ZolotukhinM commented on test/cpp/tensorexpr/test_kernel.cpp:
```diff
@@ -539,6 +539,40 @@ TEST_F(Kernel, CatInputTypesPromotion) {
+
+  TensorExprKernel k(graph);
+  StmtPtr s = k.getCodeGenStmt();
+  std::ostringstream oss;
+  oss << *s;
```
This is unneeded since we're not checking the IR

navahgar replied:
Removed

ZolotukhinM reviewed (approved):
👍

bertmaher commented:
It would be nice to have a python repro of this so it's not so tightly bound to the TE Kernel interface

bertmaher reviewed (changes requested):
Hm, actually, lots of failures, seems like there's something still broken here.

navahgar committed:
Update on "[nnc] Added a call to simplifier before inlining and updated indices during broadcast to use int64_t"


Differential Revision: [D30801686](https://our.internmc.facebook.com/intern/diff/D30801686)

[ghstack-poisoned]

navahgar commented:
> It would be nice to have a python repro of this so it's not so tightly bound to the TE Kernel interface

Added

navahgar commented:
Removed calling `simplify` before `inline`. This PR only includes the change to `LongImm` for indices during broadcast.

navahgar committed:
Update on "[nnc] Updated indices during broadcast to use int64_t"


Differential Revision: [D30801686](https://our.internmc.facebook.com/intern/diff/D30801686)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30801686).
